src/azure-cli/azure/cli/command_modules/appconfig/_client_factory.py

```diff
@@ -0,0 +1,32 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.log import get_logger
+from knack.util import CLIError
+from azure.cli.core.commands.client_factory import get_mgmt_service_client
+
+logger = get_logger(__name__)
+
+
+def get_appconfig_service_client(cli_ctx, api_version=None):
+    ''' Returns the client for managing configuration stores.'''
+    from azure.mgmt.appconfiguration import AppConfigurationManagementClient
+    return get_mgmt_service_client(cli_ctx, AppConfigurationManagementClient, api_version=api_version)
+
+
+def cf_configstore(cli_ctx, *_):
+    return get_appconfig_service_client(cli_ctx).configuration_stores
+
+
+def cf_configstore_operations(cli_ctx, *_):
+    try:
+        return get_appconfig_service_client(cli_ctx).operations
+    except CLIError:
+        params = cli_ctx.data['safe_params']
```

Comment:
You should not be using "safe_params" for any kind of logic. It is used for telemetry only.

Reply:
We are trying to bypass "az login" for some of our data plane commands by customer requests. Do you an example for how to check input arguments in _client_factory?

Reply:
You don't. Your data plane commands (which don't require login) should use a different client than your management plane commands. Storage would be the best example of this that I'm aware of.

Reply:
So, per our offline meeting, the simplest approach would be a validator on the connection string argument for your data plane commands that retrieves the connection string if it was not supplied. Then you shouldn't need two clients or any special logic in the client.

Reply:
Another way to do this is for all data plane command, we don't require "az login" if connection string is provided. Otherwise, we will try to fetch connection string based on given resource name or ID, then throw exception if it fails.